### ISSUE TITLE
Updating to Thoth.Json 3.0 breaks serialisation

### ISSUE THREAD

**Issue by rtkelly13** [thoth-org/Thoth.Json.Giraffe#1]
I'm working on a SAFE Template application v1.01.

I ran a Paket update on my project, which updated Thoth.Json.Net to v3.0.0

This is the error I got back when trying to run my application.
```
An exception of type 'System.MissingMethodException' occurred in Giraffe.dll but was not handled in user code: 'Method not found: 'System.String Auto.toString(Int32, System.Object, Microsoft.FSharp.Core.FSharpOption`1<Boolean>)'.'
   at Thoth.Json.Giraffe.ThothSerializer.Giraffe-Serialization-Json-IJsonSerializer-SerializeToBytes[T](T o)
   at Giraffe.ResponseWriters.HttpContext.WriteJsonAsync[T](HttpContext this, T dataObj)
```

I think this is due to a breaking API change that Thoth.Json.Giraffe 1.1 depends upon, when updating to Thoth.Json.Giraffe 2.0.0-beta-001 and Thoth.Json.Net 3.0 to see if that would allow a workaround that did not work.

Because of this I have rolled back to Thoth.Json.Net 2.5 in order to get it working again.

If you need me to provide a more complete Repro I am happy to provide that.

Sorry for creating an empty issue earlier not sure why it did that.

**Comment by MangelMaxime** [comment]
Hello,

when updating [SAFE-Bookstore](https://github.com/SAFE-Stack/SAFE-BookStore/) to the beta version of Thoth.Json.Giraffe we didn't see this problem.

Which code is causing this issue?

In the SAFE-Bookstore, we use the `WriteJsonAsync` at [these places](https://github.com/SAFE-Stack/SAFE-BookStore/search?q=WriteJsonAsync&unscoped_q=WriteJsonAsync)

**Comment by rtkelly13** [comment]
Managed to get an isolated repro

Run 
dotnet new SAFE -d docker -la fulma-hero --force

this is the set of changes that I got to make this error occur
https://github.com/mancfug/MancFugSafe/commit/9ff4973d29ceb6619afab249e31768a05f2d0df3

But essentially
- paket update against that deps file, (None of the other packages I have added should side effect that change, I don't think so !?!)
- getf on giraffe router return a record with a single field using the json method

Also Tried to update Thoth.Json.Giraffe to prerelease
https://github.com/mancfug/MancFugSafe/commit/4d1b8353423e1476476577345aef6a2267f707b6
The same issue occurred there too 

I can keep simplifying the repro to get it to a minimal set of changes, but this is all I have so far!

**Comment by MangelMaxime** [comment]
Thank you for reproduction.

I just read the code I still don't know what's happening yet.

That's really strange because `json` function from Saturn is implemented like that:

```fs
let json (ctx: HttpContext) (obj: 'a)   =
    ctx.WriteJsonAsync(obj)
```

Which is the same as what is done in the Bookstore sample...

@Krzysztof-Cieslak do you have an idea of what can cause the issue?

**Comment by rtkelly13** [comment]
One thing to note is that this worked by default when the counter was an int! It only started failing when a string was part of the payload! Not sure how relevant that is and how far reaching the issue goes!

**Comment by MangelMaxime** [comment]
Wait, the missing methods seems really strange: `System.String Auto.toString`

The `Auto.toString` should refers to [this member](https://github.com/thoth-org/Thoth.Json.Net/blob/master/src/Encode.fs#L482). But I wonder why it's called from `System.String`.

@alfonsogarciacaro Did we missed something when writing the auto encoder for Thoth.Json.Net?

**Comment by alfonsogarciacaro** [comment]
Hi! This is strange, I upgraded my projects to Thoth.Json 3 a few weeks ago and didn't notice anything. How can I reproduce this? I tried cloning https://github.com/mancfug/MancFugSafe/, checking out thoth-repro-issue and running `fake build -t Run` but I didn't see any error. What do you mean by "the missing method" @MangelMaxime?

**Comment by MangelMaxime** [comment]
@alfonsogarciacaro If you look at the error `'Method not found: 'System.String Auto.toString(Int32, System.Object, Microsoft.FSharp.Core.FSharpOption`1<Boolean>)`

It says `System.String Auto.toString` is not found. And when you look at the signature of the method, it looks really close to `static member toString(space : int, value : 'T, ?isCamelCase : bool, ?extra: ExtraCoders) : string =`

But all that is speculative because I didn't run the reproduction yet.

@alfonsogarciacaro Did you use the branch called `thoth-repro-issue`?

**Comment by alfonsogarciacaro** [comment]
Yes, I used the branch `thoth-repro-issue`. But where do you see the "Method not found" error? I don't see anything when running `fake build -t Run`.

<details>
 <summary>Log</summary>

```shell
MacbookPro:MancFugSafe alfonsogarciacaronunez$ fake build -t Run
Starting full restore process.
Downloading AWSSDK.Core 3.3.100.7 (Server)
Downloading Saturn 0.8 (Server)
Download of Saturn 0.8 (Server) done in 2 seconds. (403 kbit/s, 0 MB)
Download of AWSSDK.Core 3.3.100.7 (Server) done in 4 seconds. (9762 kbit/s, 5 MB)
Downloading AWSSDK.S3 3.3.101.5 (Server)
Download of AWSSDK.S3 3.3.101.5 (Server) done in 3 seconds. (10393 kbit/s, 3 MB)
Downloading AWSSDK.SQS 3.3.100.7 (Server)
Download of AWSSDK.SQS 3.3.100.7 (Server) done in 2 seconds. (3125 kbit/s, 1 MB)
run Run
Building project with version: LocalBuild
Shortened DependencyGraph for Target Run:
<== Run
   <== InstallClient
      <== Clean

The running order is:
Group - 1
  - Clean
Group - 2
  - InstallClient
Group - 3
  - Run
Starting target 'Clean'
Finished (Success) 'Clean' in 00:00:00.0120945
Starting target 'InstallClient'
Node version:
> "/Library/Frameworks/Mono.framework/Versions/Current/Commands/mono" --version (In: false, Out: true, Err: true)
/Users/alfonsogarciacaronunez/dev/MancFugSafe> "/usr/local/bin/node" --version (In: false, Out: false, Err: false)
v11.2.0
Yarn version:
/Users/alfonsogarciacaronunez/dev/MancFugSafe> "/usr/local/bin/yarn" --version (In: false, Out: false, Err: false)
1.12.1
/Users/alfonsogarciacaronunez/dev/MancFugSafe> "/usr/local/bin/yarn" install --frozen-lockfile (In: false, Out: false, Err: false)
yarn install v1.12.1
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 📃  Building fresh packages...
warning Your current version of Yarn is out of date. The latest version is "1.15.2", while you're on "1.12.1".
info To upgrade, run the following command:
$ brew upgrade yarn
✨  Done in 26.11s.
/Users/alfonsogarciacaronunez/dev/MancFugSafe/src/Client> "/usr/local/share/dotnet/dotnet" restore (In: false, Out: false, Err: false)
  Paket version 5.194.4
  The last restore is still up to date. Nothing left to do.
  Performance:
   - Runtime: 562 milliseconds
  Restore completed in 1.1 sec for /Users/alfonsogarciacaronunez/dev/MancFugSafe/src/Client/Client.fsproj.
Finished (Success) 'InstallClient' in 00:00:40.1171971
Starting target 'Run'
/Users/alfonsogarciacaronunez/dev/MancFugSafe> "/usr/local/bin/yarn" webpack-dev-server (In: false, Out: false, Err: false)
/Users/alfonsogarciacaronunez/dev/MancFugSafe/src/Server> "/usr/local/share/dotnet/dotnet" watch run (In: false, Out: false, Err: false)
yarn run v1.12.1
$ /Users/alfonsogarciacaronunez/dev/MancFugSafe/node_modules/.bin/webpack-dev-server
> "http://localhost:8080"  (In: false, Out: false, Err: false)
watch : Started
Bundling for development...
ℹ ｢wds｣: Project is running at http://localhost:8080/
ℹ ｢wds｣: webpack output is served from /
ℹ ｢wds｣: Content not from webpack is served from /Users/alfonsogarciacaronunez/dev/MancFugSafe/src/Client/public
fable-compiler 2.3.3
/Users/alfonsogarciacaronunez/dev/MancFugSafe/src/Client> dotnet restore Client.fsproj
  Paket version 5.194.4
  The last restore is still up to date. Nothing left to do.
  Performance:
   - Runtime: 445 milliseconds
  Restore completed in 386.66 ms for /Users/alfonsogarciacaronunez/dev/MancFugSafe/src/Client/Client.fsproj.


Parsing ./src/Client/Client.fsproj...
info: Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager[0]
      User profile is available. Using '/Users/alfonsogarciacaronunez/.aspnet/DataProtection-Keys' as key repository; keys will not be encrypted at rest.
Hosting environment: Production
Content root path: /Users/alfonsogarciacaronunez/dev/MancFugSafe/src/Server
Now listening on: http://0.0.0.0:8085
Application started. Press Ctrl+C to shut down.

WARNING: We noticed you're using the `useBuiltIns` option without declaring a core-js version. Currently, we assume version 2.x when no version is passed. Since this default version will likely change in future versions of Babel, we recommend explicitly setting the core-js version you are using via the `corejs` option.

You should also be sure that the version you pass to the `corejs` option matches the version specified in your `package.json`'s `dependencies` section. If it doesn't, you need to run one of the following commands:

  npm install --save core-js@2    npm install --save core-js@3
  yarn add core-js@2              yarn add core-js@3

fable: Compiled src/Client/Client.fsproj
fable: Compiled src/Client/Client.fs
fable: Compiled src/Shared/Shared.fs
fable: Compiled .fable/Fulma.1.1.0/Elements/Box.fs
fable: Compiled .fable/Fulma.1.1.0/Elements/Icon.fs
fable: Compiled .fable/Fulma.1.1.0/Elements/Notification.fs
fable: Compiled .fable/Fulma.1.1.0/Components/Card.fs
fable: Compiled .fable/Fulma.1.1.0/Elements/Heading.fs
fable: Compiled .fable/Fulma.1.1.0/Elements/Content.fs
fable: Compiled .fable/Fable.Elmish.2.0.3/cmd.fs
fable: Compiled .fable/Fulma.1.1.0/Elements/Image.fs
fable: Compiled .fable/Fable.Elmish.2.0.3/program.fs
fable: Compiled .fable/Fulma.1.1.0/Layouts/Container.fs
fable: Compiled .fable/Fulma.1.1.0/Layouts/Tile.fs
fable: Compiled .fable/Fulma.1.1.0/Layouts/Hero.fs
fable: Compiled .fable/Fulma.1.1.0/Layouts/Level.fs
fable: Compiled .fable/Fulma.1.1.0/Layouts/Columns.fs
fable: Compiled .fable/Fable.Elmish.HMR.2.1.0/hmr.fs
fable: Compiled .fable/Fable.Elmish.React.2.2.0/react.fs
fable: Compiled .fable/Fable.Elmish.Debugger.2.0.3/debugger.fs
fable: Compiled .fable/Fulma.1.1.0/Elements/Button.fs
fable: Compiled .fable/Fulma.1.1.0/Components/Navbar.fs
fable: Compiled .fable/Fable.FontAwesome.1.0.0/FontAwesome.fs
fable: Compiled .fable/Fable.PowerPack.3.0.0/Fetch.fs
fable: Compiled .fable/Fable.React.4.1.3/Fable.Helpers.React.fs
fable: Compiled .fable/Fulma.1.1.0/Layouts/Column.fs
fable: Compiled .fable/Fulma.1.1.0/Common.fs
fable: Compiled .fable/Thoth.Json.2.5.0/Decode.fs
fable: Compiled .fable/Fable.Elmish.2.0.3/prelude.fs
fable: Compiled .fable/Fable.Elmish.React.2.2.0/common.fs
fable: Compiled .fable/Fable.Elmish.Browser.2.1.0/navigation.fs
fable: Compiled .fable/Fable.Elmish.Debugger.2.0.3/Fable.Import.RemoteDev.fs
fable: Compiled .fable/Thoth.Json.2.5.0/Encode.fs
fable: Compiled .fable/Fable.PowerPack.3.0.0/Promise.fs
ℹ ｢wdm｣: Hash: c2f13ef38054d6f7c4ab
Version: webpack 4.30.0
Time: 53468ms
Built at: 04/22/2019 12:02:37 PM
     Asset       Size   Chunks             Chunk Names
    app.js   3.41 MiB      app  [emitted]  app
index.html  850 bytes           [emitted]  
  style.js   43.2 KiB    style  [emitted]  style
vendors.js   3.74 MiB  vendors  [emitted]  vendors
Entrypoint app = vendors.js app.js
Entrypoint style = vendors.js style.js
[0] multi (webpack)-dev-server/client?http://localhost:8080 (webpack)/hot/dev-server.js ./src/Client/Client.fsproj 52 bytes {app} [built]
[3] multi (webpack)-dev-server/client?http://localhost:8080 (webpack)/hot/dev-server.js ./src/Client/style.sass 52 bytes {style} [built]
[./node_modules/css-loader/dist/cjs.js!./node_modules/sass-loader/lib/loader.js?!./src/Client/style.sass] ./node_modules/css-loader/dist/cjs.js!./node_modules/sass-loader/lib/loader.js??ref--6-2!./src/Client/style.sass 1.06 KiB {style} [built]
[./node_modules/loglevel/lib/loglevel.js] 7.68 KiB {vendors} [built]
[./node_modules/querystring-es3/index.js] 127 bytes {vendors} [built]
[./node_modules/strip-ansi/index.js] 161 bytes {vendors} [built]
[./node_modules/style-loader/lib/addStyles.js] 10.2 KiB {vendors} [built]
[./node_modules/url/url.js] 22.8 KiB {vendors} [built]
[./node_modules/webpack-dev-server/client/index.js?http://localhost:8080] (webpack)-dev-server/client?http://localhost:8080 8.26 KiB {vendors} [built]
[./node_modules/webpack-dev-server/client/overlay.js] (webpack)-dev-server/client/overlay.js 3.59 KiB {vendors} [built]
[./node_modules/webpack-dev-server/client/socket.js] (webpack)-dev-server/client/socket.js 1.05 KiB {vendors} [built]
[./node_modules/webpack/hot sync ^\.\/log$] (webpack)/hot sync nonrecursive ^\.\/log$ 170 bytes {app} {style} [built]
[./node_modules/webpack/hot/dev-server.js] (webpack)/hot/dev-server.js 1.61 KiB {vendors} [built]
[./src/Client/Client.fsproj] 28 bytes {app} [built]
[./src/Client/style.sass] 1.23 KiB {style} [built]
    + 269 hidden modules
Child html-webpack-plugin for "index.html":
     1 asset
    Entrypoint undefined = index.html
    [./node_modules/html-webpack-plugin/lib/loader.js!./src/Client/index.html] 899 bytes {0} [built]
    [./node_modules/lodash/lodash.js] 527 KiB {0} [built]
    [./node_modules/webpack/buildin/global.js] (webpack)/buildin/global.js 472 bytes {0} [built]
    [./node_modules/webpack/buildin/module.js] (webpack)/buildin/module.js 497 bytes {0} [built]
ℹ ｢wdm｣: Compiled successfully.
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 GET http://localhost:8085/api/test/v1  
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
      Request finished in 1238.8306ms 200 application/json; charset=utf-8
```

</details>

**Comment by MangelMaxime** [comment]
@alfonsogarciacaro  I don't know.

I took the error from @rtkelly13 report. I didn't run is repo.

**Comment by rtkelly13** [comment]
That is interesting that you can't reproduce it @alfonsogarciacaro 

![image](https://user-images.githubusercontent.com/11708460/56506020-30dcb500-6515-11e9-94c3-9e90b1730108.png)

I hadn't tried running via fake directly to see if this was a cause but I tried that too.

<details>
  <summary>Log</summary>
  Windows PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.

Try the new cross-platform PowerShell https://aka.ms/pscore6

PS C:\git-extern\MancFugSafe-Master> fake build -t run
The last restore is still up to date. Nothing left to do.
run run
Building project with version: LocalBuild
Shortened DependencyGraph for Target Run:
<== Run
   <== InstallClient
      <== Clean

The running order is:
Group - 1
  - Clean
Group - 2
  - InstallClient
Group - 3
  - Run
Starting target 'Clean'
Finished (Success) 'Clean' in 00:00:00.0265585
Starting target 'InstallClient'
Node version:
> "C:\Program Files\Mono\bin\mono.EXE" --version (In: false, Out: true, Err: true)
C:\git-extern\MancFugSafe-Master> "C:\Program Files\nodejs\node.exe" --version (In: false, Out: false, Err: false)
v9.2.0
Yarn version:
C:\git-extern\MancFugSafe-Master> "C:\Program Files (x86)\Yarn\bin\yarn.cmd" --version (In: false, Out: false, Err: false)
1.12.3
C:\git-extern\MancFugSafe-Master> "C:\Program Files (x86)\Yarn\bin\yarn.cmd" install --frozen-lockfile (In: false, Out: false, Err: false)
yarn install v1.12.3
[1/4] Resolving packages...
success Already up-to-date.
Done in 1.07s.
C:\git-extern\MancFugSafe-Master\src\Client> "C:\Program Files\dotnet\dotnet.EXE" restore (In: false, Out: false, Err: false)
  Restore completed in 69.39 ms for C:\git-extern\MancFugSafe-Master\src\Client\Client.fsproj.
Finished (Success) 'InstallClient' in 00:00:05.8375174
Starting target 'Run'
C:\git-extern\MancFugSafe-Master> "C:\Program Files (x86)\Yarn\bin\yarn.cmd" webpack-dev-server (In: false, Out: false, Err: false)
C:\git-extern\MancFugSafe-Master\src\Server> "C:\Program Files\dotnet\dotnet.EXE" watch run (In: false, Out: false, Err: false)
yarn run v1.12.3
$ C:\git-extern\MancFugSafe-Master\node_modules\.bin\webpack-dev-server
watch : Started
> "http://localhost:8080"  (In: false, Out: false, Err: false)
info: Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager[0]
      User profile is available. Using 'C:\Users\Ryan.Kelly\AppData\Local\ASP.NET\DataProtection-Keys' as key repository and Windows DPAPI to encrypt keys at rest.
Bundling for development...
Hosting environment: Production
Content root path: C:\git-extern\MancFugSafe-Master\src\Server
Now listening on: http://0.0.0.0:8085
Application started. Press Ctrl+C to shut down.
i ｢wds｣: Project is running at http://localhost:8080/
i ｢wds｣: webpack output is served from /
i ｢wds｣: Content not from webpack is served from C:\git-extern\MancFugSafe-Master\src\Client\public
i ｢wdm｣: wait until bundle finished: /
fable-compiler 2.3.3
C:\git-extern\MancFugSafe-Master\src\Client> dotnet restore Client.fsproj
  Restore completed in 73.84 ms for C:\git-extern\MancFugSafe-Master\src\Client\Client.fsproj.


Parsing ./src/Client/Client.fsproj...

WARNING: We noticed you're using the `useBuiltIns` option without declaring a core-js version. Currently, we assume version 2.x when no version is passed. Since this default version will likely change in future versions of Babel, we recommend explicitly setting the core-js version you are using via the `corejs` option.

You should also be sure that the version you pass to the `corejs` option matches the version specified in your `package.json`'s `dependencies` section. If it doesn't, you need to run one of the following commands: 

  npm install --save core-js@2    npm install --save core-js@3
  yarn add core-js@2              yarn add core-js@3

fable: Compiled src\Client\Client.fsproj
fable: Compiled src\Client\Client.fs
fable: Compiled src\Shared\Shared.fs
fable: Compiled .fable\Fable.Elmish.2.0.3\cmd.fs
fable: Compiled .fable\Fable.Elmish.2.0.3\program.fs
fable: Compiled .fable\Fulma.1.1.0\Elements\Icon.fs
fable: Compiled .fable\Fulma.1.1.0\Components\Card.fs
fable: Compiled .fable\Fulma.1.1.0\Elements\Box.fs
fable: Compiled .fable\Fable.PowerPack.3.0.0\Fetch.fs
fable: Compiled .fable\Fulma.1.1.0\Elements\Content.fs
fable: Compiled .fable\Fulma.1.1.0\Elements\Image.fs
fable: Compiled .fable\Fable.React.4.1.3\Fable.Helpers.React.fs
fable: Compiled .fable\Fulma.1.1.0\Elements\Notification.fs
fable: Compiled .fable\Fulma.1.1.0\Elements\Heading.fs
fable: Compiled .fable\Fulma.1.1.0\Elements\Button.fs
fable: Compiled .fable\Fulma.1.1.0\Layouts\Level.fs
fable: Compiled .fable\Fulma.1.1.0\Components\Navbar.fs
fable: Compiled .fable\Fulma.1.1.0\Layouts\Container.fs
fable: Compiled .fable\Fulma.1.1.0\Layouts\Tile.fs
fable: Compiled .fable\Fable.Elmish.HMR.2.1.0\hmr.fs
fable: Compiled .fable\Fulma.1.1.0\Layouts\Hero.fs
fable: Compiled .fable\Fulma.1.1.0\Layouts\Columns.fs
fable: Compiled .fable\Fable.Elmish.React.2.2.0\react.fs
fable: Compiled .fable\Fable.FontAwesome.1.0.0\FontAwesome.fs
fable: Compiled .fable\Fable.Elmish.Debugger.2.0.3\debugger.fs
fable: Compiled .fable\Fulma.1.1.0\Common.fs
fable: Compiled .fable\Fulma.1.1.0\Layouts\Column.fs
fable: Compiled .fable\Thoth.Json.2.5.0\Decode.fs
fable: Compiled .fable\Fable.Elmish.2.0.3\prelude.fs
fable: Compiled .fable\Fable.Elmish.Debugger.2.0.3\Fable.Import.RemoteDev.fs
fable: Compiled .fable\Fable.Elmish.React.2.2.0\common.fs
fable: Compiled .fable\Thoth.Json.2.5.0\Encode.fs
fable: Compiled .fable\Fable.Elmish.Browser.2.1.0\navigation.fs
fable: Compiled .fable\Fable.PowerPack.3.0.0\Promise.fs
i ｢wdm｣: Hash: f1168e704350fcb7bdde
Version: webpack 4.30.0
Time: 40196ms
Built at: 2019-04-22 15:44:46
     Asset       Size   Chunks             Chunk Names
    app.js   3.41 MiB      app  [emitted]  app
index.html  867 bytes           [emitted]
  style.js   43.1 KiB    style  [emitted]  style
vendors.js   3.74 MiB  vendors  [emitted]  vendors
Entrypoint app = vendors.js app.js
Entrypoint style = vendors.js style.js
[0] multi (webpack)-dev-server/client?http://localhost:8080 (webpack)/hot/dev-server.js ./src/Client/Client.fsproj 52 bytes {app} [built]
[3] multi (webpack)-dev-server/client?http://localhost:8080 (webpack)/hot/dev-server.js ./src/Client/style.sass 52 bytes {style} [built]
[./node_modules/css-loader/dist/cjs.js!./node_modules/sass-loader/lib/loader.js?!./src/Client/style.sass] ./node_modules/css-loader/dist/cjs.js!./node_modules/sass-loader/lib/loader.js??ref--6-2!./src/Client/style.sass 1.06 KiB {style} [built]
[./node_modules/loglevel/lib/loglevel.js] 7.68 KiB {vendors} [built]
[./node_modules/querystring-es3/index.js] 127 bytes {vendors} [built]
[./node_modules/strip-ansi/index.js] 161 bytes {vendors} [built]
[./node_modules/style-loader/lib/addStyles.js] 10.2 KiB {vendors} [built]
[./node_modules/url/url.js] 22.8 KiB {vendors} [built]
[./node_modules/webpack-dev-server/client/index.js?http://localhost:8080] (webpack)-dev-server/client?http://localhost:8080 8.26 KiB {vendors} [built]
[./node_modules/webpack-dev-server/client/overlay.js] (webpack)-dev-server/client/overlay.js 3.59 KiB {vendors} [built]
[./node_modules/webpack-dev-server/client/socket.js] (webpack)-dev-server/client/socket.js 1.05 KiB {vendors} [built]
[./node_modules/webpack/hot sync ^\.\/log$] (webpack)/hot sync nonrecursive ^\.\/log$ 170 bytes {app} {style} [built]
[./node_modules/webpack/hot/dev-server.js] (webpack)/hot/dev-server.js 1.61 KiB {vendors} [built]
[./src/Client/Client.fsproj] 28 bytes {app} [built]
[./src/Client/style.sass] 1.23 KiB {style} [built]
    + 269 hidden modules
Child html-webpack-plugin for "index.html":
     1 asset
    Entrypoint undefined = index.html
    [./node_modules/html-webpack-plugin/lib/loader.js!./src/Client/index.html] 933 bytes {0} [built]
    [./node_modules/lodash/lodash.js] 527 KiB {0} [built]
    [./node_modules/webpack/buildin/global.js] (webpack)/buildin/global.js 472 bytes {0} [built]
    [./node_modules/webpack/buildin/module.js] (webpack)/buildin/module.js 497 bytes {0} [built]
i ｢wdm｣: Compiled successfully.
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 GET http://localhost:8085/api/test/v1
fail: Giraffe.Middleware.GiraffeErrorHandlerMiddleware[0]
      An unhandled exception has occurred while executing the request.
System.MissingMethodException: Method not found: 'System.String Auto.toString(Int32, System.Object, Microsoft.FSharp.Core.FSharpOption`1<Boolean>)'.
   at Thoth.Json.Giraffe.ThothSerializer.Giraffe-Serialization-Json-IJsonSerializer-SerializeToBytes[T](T o)
   at Giraffe.ResponseWriters.HttpContext.WriteJsonAsync[T](HttpContext this, T dataObj) in C:\projects\giraffe\src\Giraffe\ResponseWriters.fs:line 137
   at Server.getResult@35.Invoke(FSharpFunc`2 next, HttpContext ctx) in C:\git-extern\MancFugSafe-Master\src\Server\Server.fs:line 35
   at Server.action2@32.Invoke(Unit unitVar) in C:\git-extern\MancFugSafe-Master\src\Server\Server.fs:line 37
   at FSharp.Control.Tasks.TaskBuilder.run[a](FSharpFunc`2 firstStep) in C:\Users\humbo\source\repos\TaskBuilder.fs\TaskBuilder.fs:line 226
--- End of stack trace from previous location where exception was thrown ---
   at FSharp.Control.Tasks.TaskBuilder.bindTaskConfigureFalse[a,b](Task`1 task, FSharpFunc`2 continuation) in C:\Users\humbo\source\repos\TaskBuilder.fs\TaskBuilder.fs:line 127
   at FSharp.Control.Tasks.TaskBuilder.run[a](FSharpFunc`2 firstStep) in C:\Users\humbo\source\repos\TaskBuilder.fs\TaskBuilder.fs:line 226
--- End of stack trace from previous location where exception was thrown ---
   at FSharp.Control.Tasks.TaskBuilder.bindTaskConfigureFalse[a,b](Task`1 task, FSharpFunc`2 continuation) in C:\Users\humbo\source\repos\TaskBuilder.fs\TaskBuilder.fs:line 127
   at FSharp.Control.Tasks.TaskBuilder.run[a](FSharpFunc`2 firstStep) in C:\Users\humbo\source\repos\TaskBuilder.fs\TaskBuilder.fs:line 226
--- End of stack trace from previous location where exception was thrown ---
   at FSharp.Control.Tasks.TaskBuilder.bindTaskConfigureFalse[a,b](Task`1 task, FSharpFunc`2 continuation) in C:\Users\humbo\source\repos\TaskBuilder.fs\TaskBuilder.fs:line 127
   at FSharp.Control.Tasks.TaskBuilder.run[a](FSharpFunc`2 firstStep) in C:\Users\humbo\source\repos\TaskBuilder.fs\TaskBuilder.fs:line 226
--- End of stack trace from previous location where exception was thrown ---
   at FSharp.Control.Tasks.TaskBuilder.bindTaskConfigureFalse[a,b](Task`1 task, FSharpFunc`2 continuation) in C:\Users\humbo\source\repos\TaskBuilder.fs\TaskBuilder.fs:line 127
   at FSharp.Control.Tasks.TaskBuilder.run[a](FSharpFunc`2 firstStep) in C:\Users\humbo\source\repos\TaskBuilder.fs\TaskBuilder.fs:line 226
--- End of stack trace from previous location where exception was thrown ---
   at Giraffe.Middleware.Invoke@63-5.Invoke(Unit unitVar) in C:\projects\giraffe\src\Giraffe\Middleware.fs:line 63
   at FSharp.Control.Tasks.TaskBuilder.tryWith[a](FSharpFunc`2 step, FSharpFunc`2 catch) in C:\Users\humbo\source\repos\TaskBuilder.fs\TaskBuilder.fs:line 165
fail: Giraffe.Middleware.GiraffeErrorHandlerMiddleware[0]
      An unhandled exception has occurred while executing the request.
System.MissingMethodException: Method not found: 'System.String Auto.toString(Int32, System.Object, Microsoft.FSharp.Core.FSharpOption`1<Boolean>)'.
   at Thoth.Json.Giraffe.ThothSerializer.Giraffe-Serialization-Json-IJsonSerializer-SerializeToBytes[T](T o)
   at Giraffe.ResponseWriters.HttpContext.WriteJsonAsync[T](HttpContext this, T dataObj) in C:\projects\giraffe\src\Giraffe\ResponseWriters.fs:line 137
   at Server.getResult@35.Invoke(FSharpFunc`2 next, HttpContext ctx) in C:\git-extern\MancFugSafe-Master\src\Server\Server.fs:line 35
   at Server.action2@32.Invoke(Unit unitVar) in C:\git-extern\MancFugSafe-Master\src\Server\Server.fs:line 37
   at FSharp.Control.Tasks.TaskBuilder.run[a](FSharpFunc`2 firstStep) in C:\Users\humbo\source\repos\TaskBuilder.fs\TaskBuilder.fs:line 226
--- End of stack trace from previous location where exception was thrown ---
   at FSharp.Control.Tasks.TaskBuilder.bindTaskConfigureFalse[a,b](Task`1 task, FSharpFunc`2 continuation) in C:\Users\humbo\source\repos\TaskBuilder.fs\TaskBuilder.fs:line 127
   at FSharp.Control.Tasks.TaskBuilder.run[a](FSharpFunc`2 firstStep) in C:\Users\humbo\source\repos\TaskBuilder.fs\TaskBuilder.fs:line 226
--- End of stack trace from previous location where exception was thrown ---
   at FSharp.Control.Tasks.TaskBuilder.bindTaskConfigureFalse[a,b](Task`1 task, FSharpFunc`2 continuation) in C:\Users\humbo\source\repos\TaskBuilder.fs\TaskBuilder.fs:line 127
   at FSharp.Control.Tasks.TaskBuilder.run[a](FSharpFunc`2 firstStep) in C:\Users\humbo\source\repos\TaskBuilder.fs\TaskBuilder.fs:line 226
--- End of stack trace from previous location where exception was thrown ---
   at FSharp.Control.Tasks.TaskBuilder.bindTaskConfigureFalse[a,b](Task`1 task, FSharpFunc`2 continuation) in C:\Users\humbo\source\repos\TaskBuilder.fs\TaskBuilder.fs:line 127
   at FSharp.Control.Tasks.TaskBuilder.run[a](FSharpFunc`2 firstStep) in C:\Users\humbo\source\repos\TaskBuilder.fs\TaskBuilder.fs:line 226
--- End of stack trace from previous location where exception was thrown ---
   at FSharp.Control.Tasks.TaskBuilder.bindTaskConfigureFalse[a,b](Task`1 task, FSharpFunc`2 continuation) in C:\Users\humbo\source\repos\TaskBuilder.fs\TaskBuilder.fs:line 127
   at FSharp.Control.Tasks.TaskBuilder.run[a](FSharpFunc`2 firstStep) in C:\Users\humbo\source\repos\TaskBuilder.fs\TaskBuilder.fs:line 226
--- End of stack trace from previous location where exception was thrown ---
   at Giraffe.Middleware.Invoke@63-5.Invoke(Unit unitVar) in C:\projects\giraffe\src\Giraffe\Middleware.fs:line 63
   at FSharp.Control.Tasks.TaskBuilder.tryWith[a](FSharpFunc`2 step, FSharpFunc`2 catch) in C:\Users\humbo\source\repos\TaskBuilder.fs\TaskBuilder.fs:line 165
fail: Giraffe.Middleware.GiraffeErrorHandlerMiddleware[0]
      An exception was thrown attempting to handle the original exception.
System.MissingMethodException: Method not found: 'System.String Auto.toString(Int32, System.Object, Microsoft.FSharp.Core.FSharpOption`1<Boolean>)'.
   at Thoth.Json.Giraffe.ThothSerializer.Giraffe-Serialization-Json-IJsonSerializer-SerializeToBytes[T](T o)
   at Giraffe.ResponseWriters.HttpContext.WriteJsonAsync[T](HttpContext this, T dataObj) in C:\projects\giraffe\src\Giraffe\ResponseWriters.fs:line 137
   at Giraffe.Middleware.Invoke@67-8.Invoke(Unit unitVar) in C:\projects\giraffe\src\Giraffe\Middleware.fs:line 68
   at FSharp.Control.Tasks.TaskBuilder.tryWith[a](FSharpFunc`2 step, FSharpFunc`2 catch) in C:\Users\humbo\source\repos\TaskBuilder.fs\TaskBuilder.fs:line 165
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
      Request finished in 416.7672ms 500 application/json; charset=utf-8
</details>

Intrigued as to why this is specifically happening to me, unfortunately handing this laptop back in 2 days. When I get my new one happy to try and isolate the issue further and see if it was related to some specific configuration on my machine, shouldn't think so...

**Comment by rtkelly13** [comment]
I ran a git clean and it all seems to work again! The previous commit on that branch doesn't work with `git clean -xfd` then `fake build -t Run` I think it is just a breaking change for `Thoth.Json.Giraffe` and the constraints broke the integration. I will close the issue. Not sure why updating caused it to still happen when I set prerelease for `Thoth.Json.Giraffe` and it still resolved the old package at runtime
